presto-main/src/main/java/io/prestosql/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -364,10 +369,15 @@ public Expression visitValues(ValuesNode node, Void context)
 
                 List<Object> values = builder.build();
 
-                Domain domain = Domain.none(type);
-
-                if (!values.isEmpty()) {
-                    domain = domain.union(Domain.multipleValues(type, values));
+                Domain domain;
+                if (values.isEmpty()) {
+                    domain = Domain.none(type);
+                }
+                else if (hasNaN) {
```

Comment:
Consider adding comment. It might not be obvious for reader why `NaN` makes `TupleDomain` accept all non-null values.

Reply:
This knowledge is sprinkled in multiple places in the code base already.
There seems to be no good place to place such a comment though.